### PR TITLE
Template plugin tests refactoring

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Refactor template plugin tests to migrate to API 2.0
 - This is the 1st PR in a series

### Tickets:
 - [CVS-111381](https://jira.devtools.intel.com/browse/CVS-111381)

### CODE REVIEW

File: src/plugins/template/tests/functional/op_reference/comparison.hpp [hunk 1]
```diff
@@ -5,15 +5,35 @@
...
+#include "openvino/op/is_inf.hpp"
+#include "openvino/op/is_finite.hpp"
+#include "openvino/op/is_nan.hpp"
 
 using namespace ov;
 
 namespace reference_tests {
 namespace ComparisonOpsRefTestDefinitions {
 
+enum ComparisonTypes {
```
Comment (ilyachur) [inline src/plugins/template/tests/functional/op_reference/comparison.hpp:23]: As I see this enum is used in different places, should we move it to new test utils?

File: src/plugins/template/tests/functional/op_reference/conversion.hpp [hunk 2]
```diff
@@ -5,25 +5,33 @@
...
 
 using namespace ov;
 
 namespace reference_tests {
 namespace ConversionOpsRefTestDefinitions {
 
-static std::map<ngraph::helpers::ConversionTypes, std::string> conversionNames = {
-    {ngraph::helpers::ConversionTypes::CONVERT,      "Convert"},
-    {ngraph::helpers::ConversionTypes::CONVERT_LIKE, "ConvertLike"}
+enum ConversionTypes {
```
Comment (ilyachur) [inline src/plugins/template/tests/functional/op_reference/conversion.hpp:18]: The same comment

File: src/plugins/template/tests/functional/op_reference/reduction.hpp [hunk 3]
```diff
@@ -6,20 +6,56 @@
...
+        } else {
+            if (keep_dims)
+                result.push_back(1);
+        }
+    }
+
+    return result;
+}
+
+enum ReductionType {
```
Comment (ilyachur) [inline src/plugins/template/tests/functional/op_reference/reduction.hpp:41]: The same

File: src/plugins/template/tests/functional/op_reference/comparison.hpp (hunk 1 above)
Comment (olpipi) [reply src/plugins/template/tests/functional/op_reference/comparison.hpp]: done

File: src/plugins/template/tests/functional/op_reference/conversion.hpp (hunk 2 above)
Comment (olpipi) [reply src/plugins/template/tests/functional/op_reference/conversion.hpp]: done

File: src/plugins/template/tests/functional/op_reference/reduction.hpp (hunk 3 above)
Comment (olpipi) [reply src/plugins/template/tests/functional/op_reference/reduction.hpp]: done

**Review by ilyachur — APPROVED** [review]

**Pull request merged**